File: field-force-java/src/Connection/ClientSend.java
package Connection;

import AI.AI;
import Game.Move;
import Game.MoveType;
import Game.SkillType;

public class ClientSend {


    /**
     * @param packet
     * writes length of packet for appropriate form and sends packet to server
     */
    private static void sendPacket(Packet packet){
        packet.writeLength();
        Client.getInstance().sendPacket(packet);
    }

    /**
     * sends Pakettype 1. tells server that client ist playing and what name and skills were chosen
     */
    public static void sendPlayername(){
        Packet packet = new Packet();
        packet.write(ClientPackets.PLAYERNAME.getId());
        packet.write(AI.playername);
        packet.write(AI.skill1);
        packet.write(AI.skill2);
        sendPacket(packet);
    }

    /**
     * sends Pakettype 6. tells server the movereply and sets skills on cooldown if used
     */
    public static void sendMovereply(int id, Move move){
        Packet packet = new Packet();
        packet.write(ClientPackets.MOVEREPLY.getId());
        packet.write(id);
        packet.write(move);
        if(move.getType() == MoveType.SKILL) {
            move.getSkill().setOnCooldown();
            if(move.getSkill().getId() == AI.skill1)
                AI.ownPlayer.getSkill1().setOnCooldown();
            if(move.getSkill().getId() == AI.skill2)
                AI.ownPlayer.getSkill2().setOnCooldown();
        }
        sendPacket(packet);
    }
}

File: field-force-cpp/field-force-cpp/connection/clientsend.cpp
#ifdef _WIN32
#define _CRTDBG_MAP_ALLOC
#include <stdlib.h>
#include <crtdbg.h>
#ifdef _DEBUG
#ifndef DBG_NEW
#define DBG_NEW new ( _NORMAL_BLOCK , __FILE__ , __LINE__ )
#define new DBG_NEW
#endif
#endif  // _DEBUG
#endif
#include "clientsend.h"

#include "client.h"
#include "packet.h"
#include "../game/gameconstants.h"

/**
 * \brief writes length of packet for appropriate form and sends packet to server
 */
void sendPacket(Packet packet)
{
	packet.write_length();
	Client& client = Client::get_instance();
	client.send_packet(packet);
}

/**
 * \brief sends Pakettype 1. tells server that client ist playing and what name and skills were chosen
 */
void send_playername()
{
	Packet packet;
	packet.write(PLAYERNAME);
	packet.write(playername);
	packet.write(skill1);
	packet.write(skill2);
	sendPacket(packet);
}

/**
 * \brief sends Pakettype 6. tells server the movereply and sets skills on cooldown if used
 */
void send_movereply(int id, Move* move)
{
	Packet packet;
	packet.write(MOVEREPLY);
	packet.write(id);
	packet.write(*move);
	if(move->type == MoveType::MT_SKILL)
	{
		move->skill.set_on_cooldown();
	}
	sendPacket(packet);
}